Default FirebaseApp is not initialized in this process app.tivi.debug

I tried running the app on Android Studio 4.2 Canary 7 and got the following crash

**Stacktrace**

```
 --------- beginning of crash
2020-08-05 17:07:44.869 25866-25866/app.tivi.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: app.tivi.debug, PID: 25866
    java.lang.RuntimeException: Unable to create application app.tivi.TiviApplication: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process app.tivi.debug. Make sure to call FirebaseApp.initializeApp(Context) first.
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6465)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process app.tivi.debug. Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@19.3.0:184)
        at com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance(FirebaseCrashlytics.java:112)
        at app.tivi.inject.AppModule.provideFirebaseCrashlytics(AppModule.kt:130)
        at app.tivi.inject.AppModule_ProvideFirebaseCrashlyticsFactory.provideFirebaseCrashlytics(AppModule_ProvideFirebaseCrashlyticsFactory.java:36)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.getFirebaseCrashlytics(DaggerTiviApplication_HiltComponents_SingletonC.java:538)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.getTiviLogger(DaggerTiviApplication_HiltComponents_SingletonC.java:547)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.getTimberInitializer(DaggerTiviApplication_HiltComponents_SingletonC.java:551)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.getSetOfAppInitializer(DaggerTiviApplication_HiltComponents_SingletonC.java:761)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.getAppInitializers(DaggerTiviApplication_HiltComponents_SingletonC.java:765)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.injectTiviApplication2(DaggerTiviApplication_HiltComponents_SingletonC.java:1879)
        at app.tivi.DaggerTiviApplication_HiltComponents_SingletonC.injectTiviApplication(DaggerTiviApplication_HiltComponents_SingletonC.java:1865)
        at app.tivi.Hilt_TiviApplication.onCreate(Hilt_TiviApplication.java:43)
        at app.tivi.TiviApplication.onCreate(TiviApplication.kt:32)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1189)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6460)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 

```


I even tried hacking a bit like adding `FirebaseApp.initializeApp(app)` in `provideFirebaseCrashlytics` as per the suggestion but it didn't help.

```
fun provideFirebaseCrashlytics(app: Application): FirebaseCrashlytics {
        FirebaseApp.initializeApp(app)
        return FirebaseCrashlytics.getInstance()
    }
```

Not quite sure why this happened. Anyone came across this issue too?

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -17,6 +17,8 @@
 
 import app.tivi.buildsrc.Libs
 
+import java.nio.file.Files
+
 plugins {
     id 'com.android.application'
     id 'dagger.hilt.android.plugin'
@@ -242,7 +244,11 @@ android.applicationVariants.all { variant ->
     }
 }
 
-if (file("google-services.json").exists()) {
-    apply plugin: 'com.google.gms.google-services'
-    apply plugin: 'com.google.firebase.crashlytics'
+def googleServices = file("google-services.json")
+if (!googleServices.exists()) {
+    // If there isn't a google-services.json available, we'll copy our fake version to the
+    // google-services.json
+    Files.copy(file("fake-google-services.json").toPath(), googleServices.toPath())
 }
+apply plugin: 'com.google.gms.google-services'
+apply plugin: 'com.google.firebase.crashlytics'

--- a/app/fake-google-services.json
+++ b/app/fake-google-services.json
@@ -1,0 +1,85 @@
+{
+  "project_info": {
+    "project_number": "983475834758375",
+    "firebase_url": "https://fakeapp.firebaseio.com",
+    "project_id": "fake-android-app",
+    "storage_bucket": "fake-android-app.appspot.com"
+  },
+  "client": [
+    {
+      "client_info": {
+        "mobilesdk_app_id": "FAKEfakeFAKE",
+        "android_client_info": {
+          "package_name": "app.tivi"
+        }
+      },
+      "oauth_client": [
+        {
+          "client_id": "FAKEfakeFAKE",
+          "client_type": 1,
+          "android_info": {
+            "package_name": "app.tivi",
+            "certificate_hash": "FAKEfakeFAKE"
+          }
+        },
+        {
+          "client_id": "FAKEfakeFAKE",
+          "client_type": 3
+        }
+      ],
+      "api_key": [
+        {
+          "current_key": "FAKEfakeFAKE"
+        }
+      ],
+      "services": {
+        "appinvite_service": {
+          "other_platform_oauth_client": [
+            {
+              "client_id": "FAKEfakeFAKE",
+              "client_type": 3
+            }
+          ]
+        }
+      }
+    },
+    {
+      "client_info": {
+        "mobilesdk_app_id": "FAKEfakeFAKE",
+        "android_client_info": {
+          "package_name": "app.tivi.debug"
+        }
+      },
+      "oauth_client": [
+        {
+          "client_id": "FAKEfakeFAKE",
+          "client_type": 1,
+          "android_info": {
+            "package_name": "app.tivi.debug",
+            "certificate_hash": "FAKEfakeFAKE"
+          }
+        },
+        {
+          "client_id": "FAKEfakeFAKE",
+          "client_type": 3
+        }
+      ],
+      "api_key": [
+        {
+          "current_key": "FAKEfakeFAKE"
+        }
+      ],
+      "services": {
+        "appinvite_service": {
+          "other_platform_oauth_client": [
+            {
+              "client_id": "FAKEfakeFAKE",
+              "client_type": 3
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "configuration_version": "1"
+}

